### PR TITLE
Test of new implementation

### PR DESCRIPTION
**Pull request by Larssvaenjensen**

### CODE REVIEW

File: python_pellet/agent.py [hunk 1]
```diff
@@ -98,7 +102,8 @@ def qlearn(self, replay_memory, partition_memory):
 
             partitionreward[visited == 0] = 0
             partitionrewardprime[visited_prime == 0] = 0
-
+            partitionreward= torch.min(torch.stack([self.maximum_pellet_reward, partitionreward], dim=1),dim=1)[0]
```
Comment (Landi29) [inline python_pellet/agent.py:106]: Perhaps document this new partition reward calculation with a comment.

File: python_pellet/agent.py [hunk 2]
```diff
@@ -98,7 +102,8 @@ def qlearn(self, replay_memory, partition_memory):
...
+            partitionrewardprime= torch.min(torch.stack([self.maximum_pellet_reward, partitionrewardprime], dim=1),dim=1)[0]
```
Comment (Landi29) [inline python_pellet/agent.py:107]: Same as above.

File: python_pellet/environment.py [hunk 3]
```diff
@@ -90,33 +99,26 @@ def __init__(self, env):
...
-            # for Qbert sometimes we stay in lives == 0 condtion for a few frames
-            # so its important to keep lives > 0, so that we only reset once
-            # the environment advertises done.
-            done = True
-        self.lives = lives
-        return obs, reward, done, self.was_real_done
+
+        soft_terminating_state = done
+        if info['ale.lives'] < self.lives:
+            #reward -= 1
```
Comment (Landi29) [inline python_pellet/environment.py:110]: Comment or old code?

**Review by Landi29 — CHANGES_REQUESTED** [review]

File: python_pellet/environment.py (hunk 3 above)
Comment (Larssvaenjensen) [reply python_pellet/environment.py:110]: ja

File: python_pellet/agent.py [hunk 4]
```diff
@@ -98,7 +102,9 @@ def qlearn(self, replay_memory, partition_memory):
 
             partitionreward[visited == 0] = 0
             partitionrewardprime[visited_prime == 0] = 0
-
+            #these lines calculates the prelletreward as it is at this moment but with a limit to how much the pellet reward can be
```
Comment (Larssvaenjensen) [inline python_pellet/agent.py]:
```suggestion
            # these lines calculate the pellet reward as it is at this moment but with a limit to how much the pellet reward can be
```

**Review by Landi29 — APPROVED** [review]
